azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/StorageAccountApiMockTest.java

```diff
@@ -0,0 +1,288 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+import org.jclouds.azurecompute.arm.domain.Availability;
+import org.jclouds.azurecompute.arm.domain.CreateStorageServiceParams;
+import org.jclouds.azurecompute.arm.domain.StorageService;
+import org.jclouds.azurecompute.arm.domain.StorageServiceKeys;
+import org.jclouds.azurecompute.arm.domain.StorageServiceUpdateParams;
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiMockTest;
+import org.jclouds.date.DateService;
+import org.jclouds.date.internal.SimpleDateFormatDateService;
+import org.testng.annotations.Test;
+
+import java.net.MalformedURLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Test(groups = "unit", testName = "StorageAccountApiMockTest", singleThreaded = true)
+public class StorageAccountApiMockTest extends BaseAzureComputeApiMockTest {
+
+   private String subsriptionId = "SUBSCRIPTIONID";
+   private String resourceGroup = "resourceGroup";
+
+   public void testList() throws Exception {
+      server.enqueue(jsonResponse("/storageAccounts.json"));
+
+      final StorageAccountApi storageAPI = api.getStorageAccountApi(resourceGroup);
+
+      List<StorageService> list = storageAPI.list();
+      assertEquals(list, expected());
+
+      assertSent(server, "GET", "/subscriptions/" + subsriptionId +
+              "/resourcegroups/resourceGroup/providers/Microsoft.Storage/storageAccounts?api-version=2015-06-15");
+   }
+
+   public void testCreate() throws Exception {
+      server.enqueue(jsonResponse("/storageCreateResponse.json"));
+      final StorageAccountApi storageAPI = api.getStorageAccountApi(resourceGroup);
+
+      assertEquals(storageAPI.create("name-of-storage-account", "westus",
+              ImmutableMap.of("property_name", "property_value"),
+              ImmutableMap.of("accountType", StorageService.AccountType.Premium_LRS.toString())),
+              getCreateResponse());
+
+      assertSent(server, "PUT", "/subscriptions/" + subsriptionId +
+              "/resourcegroups/resourceGroup/providers/Microsoft.Storage/" +
+              "storageAccounts/name-of-storage-account?api-version=2015-06-15");
```

Comment:
For methods that generate a body we also need to verify that the generated body is the expected one.